jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/StandardDescriptorProcessor.java

```diff
@@ -1028,69 +1038,134 @@ public void addWelcomeFiles(WebAppContext context, XmlParser.Node node, Descript
         Iterator<XmlParser.Node> iter = node.iterator("url-pattern");
+        //For each url pattern, check if that has already been mapped or not
         while (iter.hasNext())
         {
-            String p = iter.next().toString(false, true);
-            p = ServletPathSpec.normalize(p);
+            String path = iter.next().toString(false, true);
+            path = ServletPathSpec.normalize(path);
 
             //check if there is already a mapping for this path
-            ListIterator<ServletMapping> listItor = _servletMappings.listIterator();
-            boolean found = false;
-            while (listItor.hasNext() && !found)
+            if (checkServletMappingPath(servletName, path, context, descriptor))
             {
-                ServletMapping sm = listItor.next();
-                if (sm.getPathSpecs() != null)
-                {
-                    for (String ps : sm.getPathSpecs())
-                    {
-                        //The same path has been mapped multiple times, either to a different servlet or the same servlet.
-                        //If its a different servlet, this is only valid to do if the old mapping was from a default descriptor.
-                        if (p.equals(ps) && (sm.isFromDefaultDescriptor() || servletName.equals(sm.getServletName())))
-                        {
-                            if (sm.isFromDefaultDescriptor())
-                            {
-                                if (LOG.isDebugEnabled())
-                                    LOG.debug("{} in mapping {} from defaults descriptor is overridden by {}", ps, sm, servletName);
-                            }
-                            else
-                                LOG.warn("Duplicate mapping from {} to {}", p, servletName);
-
-                            //remove ps from the path specs on the existing mapping
-                            //if the mapping now has no pathspecs, remove it
-                            String[] updatedPaths = ArrayUtil.removeFromArray(sm.getPathSpecs(), ps);
-
-                            if (updatedPaths == null || updatedPaths.length == 0)
-                            {
-                                if (LOG.isDebugEnabled())
-                                    LOG.debug("Removed empty mapping {}", sm);
-                                listItor.remove();
-                            }
-                            else
-                            {
-                                sm.setPathSpecs(updatedPaths);
-                                if (LOG.isDebugEnabled())
-                                    LOG.debug("Removed path {} from mapping {}", p, sm);
-                            }
-                            found = true;
-                            break;
-                        }
-                    }
-                }
+                paths.add(path);
+                context.getMetaData().setOrigin(servletName + ".servlet.mapping.url" + path, descriptor);
```

Comment:
Why not dot after the url?
```suggestion
                context.getMetaData().setOrigin(servletName + ".servlet.mapping.url." + path, descriptor);
```